sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -614,7 +622,8 @@ def build_component_spec_for_group(
         if isinstance(channel, pipeline_channel.PipelineArtifactChannel):
             component_spec.input_definitions.artifacts[
                 input_name].artifact_type.CopyFrom(
-                    type_utils.get_artifact_type_schema(channel.channel_type))
+                    type_utils.get_artifact_type_schema(channel.channel_type,
+                                                        channel.schema_version))
```

Comment:
Changes to this file only involve propagating schema_version through the compilation call stack.